### ISSUE TITLE
XMLValidator does not give error if XML declaration is not at the starting of XML

### ISSUE THREAD

**Issue by asthaaaaaa** [NaturalIntelligence/fast-xml-parser#601]
* [x ] Are you running the latest version?
* [ x] Have you included sample input, output, error, and expected output? 
* [ x] Have you checked if you are using correct configuration? 
* [ x] Did you try online [tool](https://naturalintelligence.github.io/fast-xml-parser/)?

### Input

Try to validate this XML`<note><?xml version="1.0" encoding="utf-8"?>test</note>`
### Code
```js
XMLValidator.validate(`<note><?xml version="1.0" encoding="utf-8"?>test</note>`)
```

### Output

XMLValidator returns true 

### expected data

XMLValidator.validate should return error with message `XML declaration allowed only at the start of the document.`
https://github.com/NaturalIntelligence/fast-xml-parser/blob/3c9e9fe4346a3c9924d4db45e60a7a29c565caac/src/validator.js#L206C46-L206C104

**Additional Notes:** https://github.com/NaturalIntelligence/fast-xml-parser/blob/3c9e9fe4346a3c9924d4db45e60a7a29c565caac/src/validator.js#L204 is the Root cause of the issue, as `start = i` in our case and `tagname` become `""`

**Would you like to work on this issue?**

- [x] Yes
- [ ] No

[Bookmark](https://github.com/NaturalIntelligence/fast-xml-parser/stargazers) this repository for further updates. Visit [SoloThought](https://solothought.com) to know about recent features.
